pkg/agent/iptables/iptables.go

```diff
@@ -125,10 +125,16 @@ func (c *Client) Initialize(nodeConfig *config.NodeConfig) error {
 		{FilterTable, AntreaForwardChain, []string{"-i", c.hostGateway, "!", "-o", c.hostGateway}, MarkTarget, []string{"--set-xmark", masqueradeMark}, "Antrea: mark pod to external traffic"},
 		// Accept Pod-to-external traffic which are received via host gateway interface but not sent via it.
 		{FilterTable, AntreaForwardChain, []string{"-i", c.hostGateway, "!", "-o", c.hostGateway}, AcceptTarget, nil, "Antrea: accept pod to external traffic"},
-		// Append ANTREA-POSTROUTING chain which contains Antrea related postrouting rules to POSTROUTING chain.
-		{NATTable, PostRoutingChain, nil, AntreaPostRoutingChain, nil, "Antrea: jump to Antrea postrouting rules"},
-		// Masquerade traffic requiring SNAT (has masqueradeMark set).
-		{NATTable, AntreaPostRoutingChain, []string{"-m", "mark", "--mark", masqueradeMark}, MasqueradeTarget, nil, "Antrea: masquerade traffic requiring SNAT"},
+	}
+
+	if !c.encapMode.IsNetworkPolicyOnly() {
+		// policy-only mode expects underlying network providing masquerading.
+		rules = append(rules,
+			// Append ANTREA-POSTROUTING chain which contains Antrea related postrouting rules to POSTROUTING chain.
+			rule{NATTable, PostRoutingChain, nil, AntreaPostRoutingChain, nil, "Antrea: jump to Antrea postrouting rules"},
+			// Masquerade traffic requiring SNAT (has masqueradeMark set).
+			rule{NATTable, AntreaPostRoutingChain, []string{"-m", "mark", "--mark", masqueradeMark}, MasqueradeTarget, nil, "Antrea: masquerade traffic requiring SNAT"},
```

Comment:
I have PR #462, one of whose purpose is solving the issue of cleanup of iptables when switching mode or upgrading, you may take a look and rebase on it, hoping it will get merged soon.

Reply:
sure.